upload/upload.go

```diff
@@ -175,6 +80,28 @@ func uploadFiles(files, outFiles []string, targetDir string, config *Config) err
 			return err
 		}
 
+		// get the list of files already uploaded
+		files, err := helpers.ListFiles(*config, targetDir)
+		if err != nil {
+			log.Error("Couldn't get the file list ", err)
+		}
+		if len(files.Contents) > 0 {
+
+			// range through Contents and check if we have a match
+			for l := range files.Contents {
+				// check if the two strings match
+				if aws.StringValue(files.Contents[l].Key) == config.AccessKey+"/"+targetDir+"/"+filepath.Base(filename) {
```

Comment:
```suggestion
				if aws.StringValue(files.Contents[l].Key) == config.AccessKey+path.Clean("/"+targetDir+"/")+outFiles[k] {
```

If `targetDir` is empty then there's a "//" that needs to be taken care of. 
Use of `outFiles[k]` ensures that the correct upload path of the file is considered when e.g. recursive folder upload is done.

Reply:
I added it! Also removed the loop as you suggested, only listing for each specific filename.